src/ae.c

```diff
@@ -197,6 +197,15 @@ void aeDeleteFileEvent(aeEventLoop *eventLoop, int fd, int mask)
     }
 }
 
+void aeDeleteFileEventIfNeeded(aeEventLoop *eventLoop, int fd, int mask)
+{
+    if (fd >= eventLoop->setsize) return;
+    aeFileEvent *fe = &eventLoop->events[fd];
+    if ((fe->mask & (~mask)) == fe->mask)
+        return; /* nothing to do. */
+    aeDeleteFileEvent(eventLoop, fd, mask);
+}
```

Comment:
why can't we just call `aeDeleteFileEvent`?
i fail to see where it would error in case the fd wasn't registered.

Reply:
reverted